packages/ark/src/lib/directives/ark-async.ts

```diff
@@ -0,0 +1,37 @@
+/* eslint-disable @angular-eslint/directive-class-suffix */
+import { Directive, Input } from '@angular/core';
+import { Observable, takeUntil, tap } from 'rxjs';
+
+import { ArkAsyncBase } from './ark-async-base';
+
+/**
+ * Structural directive replacing `async` pipe. Accepts observable,
+ * automatically subscribes to it, unsubscribes from it on destroy,
+ * and provides context variable that contains emitted value.
+ * Every time the value emits, the directive detects changes inside its scope.
+ *
+ * Example usage:
+ * ```
+ * <ng-container *arkAsync="let time; from: time$">
+ *   <span>{{ time | date : 'HH:mm:ss' }}</span>
+ * </ng-container>
+ * ```
+ */
+@Directive({
+  selector: '[arkAsync],[arkAsyncFrom]',
+  standalone: true,
+})
+export class ArkAsync<T = unknown> extends ArkAsyncBase<T> {
+  @Input()
+  set arkAsyncFrom(obs$: Observable<T>) {
+    obs$
```

Comment:
I think there's a mem-leak here, keep the subscription and guard this line by unsubscribing first before subscribing again

Reply:
🛠️ fixed

Reply:
The setter can `next(obs$)` on some subject, and that subject can be the source of the entire pipe. That way you can skip implementing `ngOnChanges` and the manual call unsubscribe.

```typescript
private newObservable = new Subject<Observable<T>>();

set arkAsyncFrom(obs$) {
   this.newObservable.next(obs$);
}

constructor() {
    this.newObservable.pipe(
        takeUntil(this.destroyed),
        switchMap(obs => obs.pipe(takeUntil(this.destroyed))
        .....
    ).subscribe(....)
}
```

Just a suggestion :)

Reply:
@BeSpunky this is something that I call a setter bomb. I use `ngOnChanges` on purpose because it's the only way to be sure that all setters are set. Maybe I'll write an article about it 😊